Java - UnsatisfiedLinkError

### What happened?

I've followed the [quickstart guide](https://github.com/columnar-tech/adbc-quickstarts/tree/main/java/clickhouse#readme) for Java / ClickHouse but am seeing the following error:

```
Exception in thread "main" java.lang.UnsatisfiedLinkError: C:\Users\Edward\AppData\Local\Temp\adbc-jni-5247101362949043213.tmp: Can't find dependent libraries
        at java.base/jdk.internal.loader.NativeLibraries.load(Native Method)
        at java.base/jdk.internal.loader.NativeLibraries$NativeLibraryImpl.open(NativeLibraries.java:388)
        at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:232)
        at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:174)
        at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2394)
        at java.base/java.lang.Runtime.load0(Runtime.java:755)
        at java.base/java.lang.Runtime.load(Runtime.java:741)
        at org.apache.arrow.adbc.driver.jni.impl.JniLoader.<init>(JniLoader.java:55)
        at org.apache.arrow.adbc.driver.jni.impl.JniLoader.<clinit>(JniLoader.java:34)
        at org.apache.arrow.adbc.driver.jni.JniDriver.open(JniDriver.java:61)
        at org.apache.arrow.adbc.drivermanager.AdbcDriverManager.connect(AdbcDriverManager.java:74)
        at tech.columnar.Example.main(Example.java:42)
[ERROR] Command execution failed.
org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
    at org.apache.commons.exec.DefaultExecutor.executeInternal (DefaultExecutor.java:394)
    at org.apache.commons.exec.DefaultExecutor.execute (DefaultExecutor.java:292)
```

On line 35 of Example.java, I've tried replacing `"clickhouse"` with a path to the driver .dll but the same issue occurs. I've seen this across two different machines so hopefully not user error.

Thanks

I've also raised this here - https://github.com/columnar-tech/adbc-quickstarts/issues/117

### Stack Trace

_No response_

### How can we reproduce the bug?

_No response_

### Environment/Setup

Windows 11
Java 17
Latest ClickHouse Docker img

--- a/.github/workflows/packaging.yml
+++ b/.github/workflows/packaging.yml
@@ -284,6 +284,7 @@ jobs:
         working-directory: adbc
         env:
           ADBC_BUILD_STATIC: ON
+          ADBC_BUILD_SHARED: OFF
           ADBC_BUILD_TESTS: OFF
           ADBC_USE_ASAN: OFF
           ADBC_USE_UBSAN: OFF

